src/boot/efi/boot.c

```diff
@@ -1877,15 +1913,23 @@ static BOOLEAN config_entry_add_call(
 
         entry = xnew(ConfigEntry, 1);
         *entry = (ConfigEntry) {
-                .id = xstrdup(id),
-                .title = xstrdup(title),
+                .type = type,
                 .call = call,
                 .tries_done = UINTN_MAX,
                 .tries_left = UINTN_MAX,
         };
 
+        if (type == LOADER_SECURE_BOOT_KEYS) {
+                entry->id = xpool_print(L"secure-boot-keys-%s", id);
+                entry->title = xpool_print(L"Enroll Secure Boot keys: %s", title);
```

Comment:
why do this in the generic config_entry_add_call() function? why not pass the right values right-away?

(we don't really want to optimize memory allocations away, outside of inner loops. i.e. we prefer readability, uniformity and encapsulation over the absolute minimization of memory allocations)